targets/riot-stm32f4/README.md

```diff
@@ -1,96 +1,139 @@
+[RIOT](https://www.riot-os.org/)
 
 ### How to build
 
-#### 1. Preface
+#### 1. Setup the build environment for STM32F4-Discovery board
 
-1, Directory structure
+Clone the necessary projects into a `jerry-riot` directory.
+The latest tested working version of RIOT is `2021.10`.
 
-Assume `harmony` as the path to the projects to build.
-The folder tree related would look like this.
+```sh
+# Create a base folder for all the projects.
+mkdir jerry-riot && cd jerry-riot
 
+git clone https://github.com/jerryscript-project/jerryscript.git
+git clone https://github.com/RIOT-OS/RIOT.git -b 2021.10
 ```
-harmony
+
+The following directory structure has been created:
+
+```
+jerry-riot
   + jerryscript
   |  + targets
-  |      + riot-stm32f4
+  |     + riot-stm32f4
   + RIOT
 ```
 
-2, Target board
+#### 2. Build JerryScript
 
-Assume [STM32F4-Discovery with BB](http://www.st.com/web/en/catalog/tools/FM116/SC959/SS1532/LN1199/PF255417)
-as the target board.
+```sh
+# Assuming you are in jerry-riot folder.
+jerryscript/tools/build.py \
+    --clean \
+    --lto=OFF \
+    --jerry-cmdline=OFF \
+    --jerry-math=ON \
+    --amalgam=ON \
+    --mem-heap=70 \
+    --profile=es.next \
+    --compile-flag=-fshort-enums \
+    --toolchain=${PWD}/jerryscript/cmake/toolchain_mcu_stm32f4.cmake
+```
 
-#### 2. Prepare RIOT-OS
+#### 3. Build RIOT
 
-Follow [this](https://www.riot-os.org/#download) page to get the RIOT-OS source.
+```
+# Assuming you are in jerry-riot folder.
+make BOARD=stm32f4discovery -f jerryscript/targets/riot-stm32f4/Makefile
+```
 
-Follow the [Inroduction](https://github.com/RIOT-OS/RIOT/wiki/Introduction) wiki site and also check that you can flash the stm32f4-board.
+The created binary is a `riot_jerryscript.elf` named file located in `jerryscript/build/riot-stm32f4/bin/` folder.
 
+#### 4. Flash the device
 
-#### 3. Build JerryScript for RIOT-OS
+Connect Mini-USB for charging and flashing the device. Install `openocd` tool that is used by RIOT.
 
 ```
-# assume you are in harmony folder
-cd jerryscript
-make -f ./targets/riot-stm32f4/Makefile.riot
+# Assuming you are in jerry-riot folder.
+make BOARD=stm32f4discovery -f jerryscript/targets/riot-stm32f4/Makefile flash
 ```
 
-This will generate the following libraries:
-```
-/build/bin/release.riotstm32f4/librelease.jerry-core.a
-/build/bin/release.riotstm32f4/librelease.jerry-math.lib.a
-```
+Note: `ST-LINK` also can be used that is described at [this page](https://github.com/RIOT-OS/RIOT/wiki/ST-LINK-tool).
+
+### 5. Connect to the device
+
+Use `USB To TTL Serial Converter` for serial communication. STM32F4-Discovery pins are mapped by RIOT as follows:
 
-This will copy one library files to `targets/riot-stm32f4/bin` folder:
 ```
-libjerrycore.a
+  PA2 = TX Data
+  PA3 = RX Data
+  GND = GND
 ```
 
-This will create a hex file in the `targets/riot-stm32f4/bin` folder:
-```
-riot_jerry.elf
+The device should be visible as `/dev/ttyUSB0`.
+
+#### 5.1. Serial communication by `pyterm`
```

Comment:
From previous experience, pyterm is often flakey. Please consider leaving only the minicom in the description.